packages/framework/src/parser.ts

```diff
@@ -46,22 +47,28 @@ export interface EnsembleWidgetYAML {
 
 export const EnsembleParser = {
   parseApplication: (app: ApplicationDTO): EnsembleAppModel => {
+    const customWidgets = (app.widgets ?? []).map(({ name, content: yaml }) =>
+      EnsembleParser.parseWidget(name, parse(yaml) as EnsembleWidgetYAML),
+    );
+
+    const widgetApis: EnsembleAPIModel[] = flatMap(
+      customWidgets,
+      (widget) => widget.apis ?? [],
+    );
+
     const screens = app.screens.map(({ name, content: yaml }) => {
       const screen = parse(yaml) as EnsembleScreenYAML;
       const viewGroup = get(screen, "ViewGroup");
       if (viewGroup) {
         return EnsembleParser.parseMenu(viewGroup);
       }
-      return EnsembleParser.parseScreen(name, screen, app);
+
+      return EnsembleParser.parseScreen(name, screen, app, widgetApis);
```

Comment:
Concat after instead of introducing another argument

Reply:
@evshi are you suggesting like, 
`return EnsembleParser.parseScreen(name, screen, {app, widgetApis});`